python/src/iceberg/transforms.py

```diff
@@ -301,6 +301,91 @@ def _(self, value_type: IcebergType, value: int) -> str:
         return datetime.to_human_timestamptz(value)
 
 
+class TruncateTransform(Transform[S, S]):
+    """A transform for truncating a value to a specified width.
+    Args:
+      source_type (Type): An Iceberg Type of IntegerType, LongType, StringType, BinaryType or DecimalType
+      width (int): The truncate width
+    Raises:
+      ValueError: If a type is provided that is incompatible with a Truncate transform
+    """
+
+    def __init__(self, source_type: IcebergType, width: int):
+        super().__init__(
+            f"truncate[{width}]",
+            f"transforms.truncate(source_type={repr(source_type)}, width={width})",
+        )
+        self._type = source_type
+        self._width = width
+
+    @property
+    def width(self) -> int:
+        return self._width
+
+    @property
+    def type(self) -> IcebergType:
+        return self._type
+
+    def apply(self, value: Optional[S]) -> Optional[S]:
+        return self._truncate_value(value) if value is not None else None
```

Comment:
Nit: I'm just trying to popularize the Walrus operator:
```suggestion
        return truncated if (truncated := self._truncate_value(value)) else None
```

Reply:
Why not just `return self._truncate_value(value)`? If that's `None`, then returning it will return `None`.

Reply:
Actually, I see why: the `_truncate_value` method doesn't use `Optional`. So it should only be called if `value` is not `None`. I think the original is correct.

Reply:
Wondering the benefit to use Walrus operator in this case. It seems unnecessarily add a new variable `truncated`.